scenes/game_logic/fill_game_logic.gd

```diff
@@ -25,6 +26,10 @@ func start() -> void:
 func _ready() -> void:
 	var filling_barrels: Array = get_tree().get_nodes_in_group("filling_barrels")
 	barrels_to_win = clampi(barrels_to_win, 0, filling_barrels.size())
+	if intro_dialogue:
+		DialogueManager.show_dialogue_balloon(intro_dialogue, "", [self])
```

Comment:
This is an existing issue but I think we might in future want to pass the player in as extra state, in case the dialogue has them speaking.